code/src/main/java/habit_mode/model/sudoku/SudokuPuzzle.java

```diff
@@ -0,0 +1,190 @@
+package habit_mode.model.sudoku;
+
+import java.util.Stack;
+
+/**
+ * Sudoku puzzle class
+ * 
+ * @author Matthew Thompson
+ * @version spring 2022
+ */
+public class SudokuPuzzle {
+
+    private int[][] numbers;
+    private boolean[][] numberLocks;
+    private int[][] defaultAnswer;
+    private int selectedNumber;
+    private Stack<SudokuMove> moveHistory;
+
+    /**
+     * Default Constructor
+     * @precondition none
+     * @postcondition this.numbers = new int[9][9] && this.numberLocks = new boolean[9][9] && this.defaultAnswer = new int[9][9] && this.selectedNumber = -1 && this.moveHistory = new Stack<SudokuMove>()
+     */
+    public SudokuPuzzle() {
+        this.numbers = new int[9][9];
+        this.numberLocks = new boolean[9][9];
+        this.defaultAnswer = new int[9][9];
+        this.selectedNumber = -1;
+        this.moveHistory = new Stack<SudokuMove>();
+    }
+
+    /**
+     * Gets the numbers
+     * @precondition none
+     * @postcondition none
+     * @return this.numbers
+     */
+    public int[][] getNumbers() {
+        return this.numbers;
+    }
+
+    /**
+     * Gets the move history
+     * @precondition none
+     * @postcondition none
+     * @return this.moveHistory
+     */
+    public Stack<SudokuMove> getMoveHistory() {
+        return this.moveHistory;
+    }
+
+    /**
+     * Gets the selected number
+     * @precondition none
+     * @postcondition none
+     * @return selected number
+     */
+    public int getSelectedNumber() {
+        return this.selectedNumber;
+    }
+
+    /**
+     * Gets the number at specified location
+     * 
+     * @param column the column
+     * @param row    the row
+     * @precondition 0 <= column <= 8 && 0 <= row <= 8
+     * @return number at specified position
+     */
+    public int getNumber(int column, int row) {
+        this.checkColumnAndRow(column, row);
+        return this.numbers[row][column];
+    }
+
+    private void checkColumnAndRow(int column, int row) {
+        if (column < 0) {
+            throw new IndexOutOfBoundsException("column can not be less than 0");
+        }
+        if (row < 0) {
+            throw new IndexOutOfBoundsException("row can not be less than 0");
+        }
+    }
+
+    /**
+     * Gets the number locks
+     * @precondition none
+     * @postcondition none
+     * @return this.numberLocks
```

Comment:
We can't see what a private field is from the outside. The return should not mention it, and should tell the developer what the return value is in words.